daemon/snapper/snapper.go

```diff
@@ -257,21 +278,62 @@ func snapshot(a args, u updater) state {
 			progress.state = SnapStarted
 		})
 
-		l.Debug("create snapshot")
-		err := zfs.ZFSSnapshot(fs, snapname, false)
-		if err != nil {
-			hadErr = true
-			l.WithError(err).Error("cannot create snapshot")
+		var doneAt time.Time
+
+		var snapErr error
+		var preHookErr []error
+		var postHookErr []error
+		hookEnvExtra := map[string]string{
+			"fs":       fs.ToString(),
```

Comment:
I left it separated, thinking of future hooks like a post-job hook where a single `fs` name wouldn't apply.

Reply:
Ah, I see:

https://github.com/zrepl/zrepl/pull/203/files#diff-dce3c966e41dcdc41961b89f87185a4cR117

I think the naked "fs" and later 'magic' conversion into `ZREPL_FS` makes it actually harder to understand where `ZREPL_FS` is set, e.g. when grepping the source code.
I would suggest extracting all ZREPL_ environment variables into constants `hooks.Env{Hooktype,Phase,DryRun,FS,...}` of the hooks package, including `ZREPL_FS`. Then just use `hooks.EnvFS` instead of the naked `fs` and get rid of the 'magic' logic.